utilities/post_migration.py

```
@@ -708,6 +708,48 @@ def check_network(source_vm: dict[str, Any], destination_vm: dict[str, Any], net
+        destination_vm: Destination VM data from vm_dict(), containing
+            network_interfaces with guest_interface_name field
+
+    Raises:
+        ValueError: If source or destination has no network interfaces
+        AssertionError: If any NIC name doesn't match between source and destination
+    """
+    source_nics = source_vm_data.get("network_interfaces", [])
+    dest_nics = destination_vm.get("network_interfaces", [])
+
+    if not source_nics:
+        raise ValueError("Source VM has no network interfaces — cannot verify NIC name preservation")
```

Ruff (0.15.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)